src/lmlog/intelligence/aggregation.py

```
@@ -0,0 +1,596 @@
+        normalized = message
+        variables = {}
+
+        for pattern, placeholder in self._variable_patterns:
+            matches = list(pattern.finditer(normalized))
+            for i, match in enumerate(reversed(matches)):
+                var_name = f"{placeholder}_{i}"
+                variables[var_name] = match.group()
+                normalized = (
+                    normalized[: match.start()]
+                    + placeholder
+                    + normalized[match.end() :]
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)